### ISSUE TITLE
How to build h2o binary only

### ISSUE THREAD

**Issue by montaniasystemab** [h2o/h2o#627]
I've been trying to build h2o as a binary only, without libh2o. I've not found any instructions and my cmake / make knowledge doesn't seem to be enough to figure it out. 

`make h2o` seems to build only the binary, but `make install` will try to build libh2o before installing: 

```
# make h2o
[  0%] Built target bundled-ssl
[100%] Built target h2o

# make install DESTDIR=/tmp/test
[  0%] Built target bundled-ssl
[ 37%] Built target h2o
[ 37%] Building C object CMakeFiles/libh2o.dir/deps/cloexec/cloexec.c.o
[ 37%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/close.c.o
[ 39%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/connect.c.o
[ 39%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/recv.c.o
[ 39%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/send.c.o
[ 41%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/socket.c.o
[ 41%] Building C object CMakeFiles/libh2o.dir/deps/libyrmcds/strerror.c.o
[ 41%] Building C object CMakeFiles/libh2o.dir/deps/picohttpparser/picohttpparser.c.o
[ 42%] Building C object CMakeFiles/libh2o.dir/lib/common/file.c.o
[ 42%] Building C object CMakeFiles/libh2o.dir/lib/common/filecache.c.o
[ 42%] Building C object CMakeFiles/libh2o.dir/lib/common/hostinfo.c.o
In file included from /usr/src/h2o/h2o-1.6.0/include/h2o/socket.h:60,
                 from /usr/src/h2o/h2o-1.6.0/include/h2o/multithread.h:27,
                 from /usr/src/h2o/h2o-1.6.0/include/h2o/hostinfo.h:32,
                 from /usr/src/h2o/h2o-1.6.0/lib/common/hostinfo.c:22:
/usr/src/h2o/h2o-1.6.0/include/h2o/socket/uv-binding.h:28:2: error: #error "libh2o (libuv binding) requires libuv version 1.x.y"
make[2]: *** [CMakeFiles/libh2o.dir/lib/common/hostinfo.c.o] Error 1
make[1]: *** [CMakeFiles/libh2o.dir/all] Error 2
make: *** [all] Error 2
```

I've seen libuv mentioned as a optional dependency. So how can I build and install h2o without it?

Thanks!

**Comment by gaoyichuan** [comment]
h2o is completely dependent to libh2o. So there's no way to do that..

**Comment by kazuho** [comment]
@montaniasystemab Can you rerun cmake on an empty directory and post the full output?

`make install` (or `make all` which is called by `make install`) is designed to build libh2o only when libuv is found.  However, in your case it is likely that an older version (earlier than 1.0.0) of libuv is mis-detected by CMake, and I would like to know what is happening.

**Comment by montaniasystemab** [comment]
Yes, libuv 0.10 is installed from epel:

```
# yum list installed | grep libuv
libuv.x86_64                         1:0.10.34-1.el6                   @epel
libuv-devel.x86_64                   1:0.10.34-1.el6                   @epel
```

Output from these commands: 
```
cmake -DWITH_BUNDLED_SSL=on .
make h2o
make install DESTDIR=/tmp/$DIR
```

https://gist.github.com/montaniasystemab/82696d2e5057e440d006

System i Centos 6

**Comment by kazuho** [comment]
@montaniasystemab Thank you for the log.  It helps a lot!

Looking at the log and CMakeLists.txt, I think I have spotted the cause / have created a fix in #630.

May I ask you to try if applying the PR on #630 fixes the issue?

**Comment by montaniasystemab** [comment]
@kazuho It works! Thanks for the quick response and fix

```
# /tmp/h2o/usr/local/bin/h2o --version
h2o version 1.7.0-alpha1
OpenSSL: LibreSSL 2.2.4
```
